xdsl/dialects/x86/ops.py

```diff
@@ -62,6 +62,27 @@
 R3InvT = TypeVar("R3InvT", bound=X86RegisterType)
 
 
+def parse_memory_access_offset(parser: Parser) -> IntegerAttr:
+    offset_type = IntegerType(64, Signedness.SIGNED)
+    temp = parse_optional_immediate_value(
+        parser,
+        offset_type,
+    )
+    if temp is None:
+        temp = IntegerAttr(0, offset_type)
+    assert isinstance(temp, IntegerAttr)
+    return temp
+
+
+def parse_comma_memory_access_offset(parser: Parser) -> IntegerAttr:
```

Comment:
This seems like something that would make more sense inline, let's take a look at this helper together later?